include/bluetooth/iso.h

```diff
@@ -0,0 +1,256 @@
+/** @file
+ *  @brief Bluetooth ISO handling
+ */
+
+/*
+ * Copyright (c) 2020 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#ifndef ZEPHYR_INCLUDE_BLUETOOTH_ISO_H_
+#define ZEPHYR_INCLUDE_BLUETOOTH_ISO_H_
+
+/**
+ * @brief ISO
+ * @defgroup bt_iso ISO
+ * @ingroup bluetooth
+ * @{
+ */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <sys/atomic.h>
+#include <bluetooth/buf.h>
+#include <bluetooth/conn.h>
+#include <bluetooth/hci.h>
+
+/** @def BT_ISO_CHAN_SEND_RESERVE
+ *  @brief Headroom needed for outgoing buffers
+ */
+#define BT_ISO_CHAN_SEND_RESERVE (CONFIG_BT_HCI_RESERVE + \
+				  BT_HCI_ISO_HDR_SIZE + \
+				  BT_HCI_ISO_DATA_HDR_SIZE)
+
+struct bt_iso_chan;
+
+/** @brief Life-span states of ISO channel. Used only by internal APIs
+ *  dealing with setting channel to proper state depending on operational
+ *  context.
+ */
+enum {
+	/** Channel disconnected */
+	BT_ISO_DISCONNECTED,
+	/** Channel bound to a connection */
+	BT_ISO_BOUND,
+	/** Channel in connecting state */
+	BT_ISO_CONNECT,
+	/** Channel ready for upper layer traffic on it */
+	BT_ISO_CONNECTED,
+	/** Channel in disconnecting state */
+	BT_ISO_DISCONNECT,
+};
+
+/** @brief ISO Channel structure. */
+struct bt_iso_chan {
+	/** Channel connection reference */
+	struct bt_conn			*conn;
+	/** Channel operations reference */
+	struct bt_iso_chan_ops		*ops;
+	/** Channel QoS reference */
+	struct bt_iso_chan_qos		*qos;
+	/** Channel data path reference*/
+	struct bt_iso_chan_path		*path;
+	sys_snode_t			node;
+	uint8_t				state;
+	bt_security_t			required_sec_level;
+};
+
+/** @brief Audio QoS direction */
+enum {
+	BT_ISO_CHAN_QOS_IN,
+	BT_ISO_CHAN_QOS_OUT,
+	BT_ISO_CHAN_QOS_INOUT
+};
+
+/** @brief ISO Channel QoS structure. */
+struct bt_iso_chan_qos {
+	/** @brief Channel direction
+	 *
+	 *  Possible values: BT_ISO_CHAN_QOS_IN, BT_ISO_CHAN_QOS_OUT or
+	 *  BT_ISO_CHAN_QOS_INOUT.
+	 */
+	uint8_t				dir;
+	/** Channel interval */
+	uint32_t			interval;
+	/** Channel SCA */
+	uint8_t				sca;
+	/** Channel packing mode */
+	uint8_t				packing;
+	/** Channel framing mode */
```

Comment:
But what does it mean? E.g. should I set it to 23?

In general these comments don't seem to stand on their own

Reply:
It is the framing mode from the spec, either framed or unframed, the upper layer will have to decide in which mode it shall be used.